language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ConvertToQueryExpressionCodeAction.java

```diff
@@ -176,65 +180,73 @@ private Optional<LhsRhsSymbolInfo> getLhsAndRhsSymbolInfo(NonTerminalNode matche
+            } else if (matchedNode.kind() == SyntaxKind.SPECIFIC_FIELD) {
+                SpecificFieldNode specificFieldNode = (SpecificFieldNode) matchedNode;
+                lhsNode = specificFieldNode.fieldName();
+                rhsNode = specificFieldNode.valueExpr().get();
                 lhsSymbol = semanticModel.symbol(lhsNode)
-                        .filter(symbol -> symbol.kind() == SymbolKind.VARIABLE);
-                lhsType = lhsSymbol.map(symbol -> ((VariableSymbol) symbol).typeDescriptor());
-            }
-        } else if (matchedNode.kind() == SyntaxKind.ASSIGNMENT_STATEMENT) {
-            // There can be 2 types here: Variable assignments and field access expressions.
-            // 1. list = otherList;
-            // 2. obj.list = otherList;
-            AssignmentStatementNode assignmentStatementNode = (AssignmentStatementNode) matchedNode;
-            if (assignmentStatementNode.varRef().kind() == SyntaxKind.FIELD_ACCESS) {
-                FieldAccessExpressionNode exprNode = (FieldAccessExpressionNode) assignmentStatementNode.varRef();
-                lhsNode = exprNode.fieldName();
-                lhsSymbol = semanticModel.symbol(lhsNode)
-                        .filter(symbol -> symbol.kind() == SymbolKind.CLASS_FIELD
-                                || symbol.kind() == SymbolKind.RECORD_FIELD);
-                lhsType = lhsSymbol
-                        .map(symbol -> symbol.kind() == SymbolKind.CLASS_FIELD ?
-                                ((ClassFieldSymbol) symbol).typeDescriptor()
-                                : ((RecordFieldSymbol) symbol).typeDescriptor());
-            } else {
-                lhsNode = assignmentStatementNode.varRef();
-                lhsSymbol = semanticModel.symbol(lhsNode)
-                        .filter(symbol -> symbol.kind() == SymbolKind.VARIABLE);
+                        .filter(symbol -> symbol.kind() == SymbolKind.RECORD_FIELD);
                 lhsType = lhsSymbol
-                        .map(symbol -> ((VariableSymbol) symbol).typeDescriptor());
+                        .map(symbol -> ((RecordFieldSymbol) symbol).typeDescriptor());
+                break;
             }
-            rhsNode = assignmentStatementNode.expression();
-        } else if (matchedNode.kind() == SyntaxKind.SPECIFIC_FIELD) {
-            SpecificFieldNode specificFieldNode = (SpecificFieldNode) matchedNode;
-            lhsNode = specificFieldNode.fieldName();
-            rhsNode = specificFieldNode.valueExpr().get();
-            lhsSymbol = semanticModel.symbol(lhsNode)
-                    .filter(symbol -> symbol.kind() == SymbolKind.RECORD_FIELD);
-            lhsType = lhsSymbol
-                    .map(symbol -> ((RecordFieldSymbol) symbol).typeDescriptor());
-        }
-        
+            matchedNode = matchedNode.parent();
+        } while (matchedNode.kind() != SyntaxKind.LOCAL_VAR_DECL ||
+                matchedNode.kind() != SyntaxKind.ASSIGNMENT_STATEMENT
```

Comment:
Shouldn't we add a check like `matchedNode != null` too?

Reply:
If the matchedNode was null the code action would not be invoked. See https://github.com/ballerina-platform/ballerina-lang/blob/05c1101887657fe5cf4b5119676fb74e8c4cded1/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionRouter.java#L81

Reply:
But we modify that inside the `while` loop `matchedNode = matchedNode.parent();` right?

Reply:
+1 added the the check